File: clients/cpp/MyStrategy.cpp
#include "MyStrategy.hpp"
#include <exception>

MyStrategy::MyStrategy(const model::Constants& constants) {}

model::Order MyStrategy::getOrder(const model::Game& game, DebugInterface* debugInterface)
{
    return model::Order(std::unordered_map<int, model::UnitOrder>());
}

void MyStrategy::debugUpdate(DebugInterface& debugInterface) {}

void MyStrategy::finish() {}

File: clients/php/MyStrategy.php
<?php

require_once 'Model/Game.php';
require_once 'Model/Order.php';
require_once 'Model/Constants.php';

class MyStrategy
{
    function __construct(\Model\Constants $constants) {}
    function getOrder(\Model\Game $game, ?DebugInterface $debugInterface): \Model\Order
    {
        return new \Model\Order([]);
    }
    function debugUpdate(DebugInterface $debug_interface) {}
    function finish() {}
}